ComponentAssemblies/ClientComponents/GaugeWidgets/clients/com/vantiq/clientcomponent/Thermometer.json

```
@@ -0,0 +1,162 @@
+        "functionParameters" : [ "value" ],
+        "hidden" : true,
+        "propertyLabel" : "drawTemp",
+        "propertyType" : 62,
+        "type" : 1
+      }, {
+        "defaultValue" : " \n  const startValue = configuration.current; // The current battery level\n  const changeInValue = targetTemp - startValue;\n  const startTime = performance.now();\n\n  function easeOutQuad(t) {\n    return t * (2 - t);\n  }\n  \n  function animate(currentTime) {\n    // Determine the time elapsed since the start of the animation\n    const elapsedTime = currentTime - startTime;\n    // Determine the proportion of the duration that has elapsed (clamped between 0 and 1)\n    const timeFraction = Math.min(elapsedTime / duration, 1);\n    // Apply the easing function to get the current progress\n    const easing = easeOutQuad(timeFraction);\n    // Calculate the current battery level using easing\n    const currentValue = startValue + changeInValue * easing;\n\n    // Update the battery level by calling setBattery\n    configuration.setTemp(currentValue);\n\n    // If the animation is not yet complete, request the next animation frame\n    if (timeFraction < 1) {\n      requestAnimationFrame(animate);\n    }\n  }\n\n  requestAnimationFrame(animate);\n  configuration.current = targetTemp;\n",
```

setBattery in comment should be setTemp
```suggestion
        "defaultValue" : " \n  const startValue = configuration.current; // The current battery level\n  const changeInValue = targetTemp - startValue;\n  const startTime = performance.now();\n\n  function easeOutQuad(t) {\n    return t * (2 - t);\n  }\n  \n  function animate(currentTime) {\n    // Determine the time elapsed since the start of the animation\n    const elapsedTime = currentTime - startTime;\n    // Determine the proportion of the duration that has elapsed (clamped between 0 and 1)\n    const timeFraction = Math.min(elapsedTime / duration, 1);\n    // Apply the easing function to get the current progress\n    const easing = easeOutQuad(timeFraction);\n    // Calculate the current battery level using easing\n    const currentValue = startValue + changeInValue * easing;\n\n    // Update the battery level by calling setTemp\n    configuration.setTemp(currentValue);\n\n    // If the animation is not yet complete, request the next animation frame\n    if (timeFraction < 1) {\n      requestAnimationFrame(animate);\n    }\n  }\n\n  requestAnimationFrame(animate);\n  configuration.current = targetTemp;\n",
```

---

and "battery" in 3 comments should be "thermometer"
```suggestion
        "defaultValue" : " \n  const startValue = configuration.current; // The current thermometer level\n  const changeInValue = targetTemp - startValue;\n  const startTime = performance.now();\n\n  function easeOutQuad(t) {\n    return t * (2 - t);\n  }\n  \n  function animate(currentTime) {\n    // Determine the time elapsed since the start of the animation\n    const elapsedTime = currentTime - startTime;\n    // Determine the proportion of the duration that has elapsed (clamped between 0 and 1)\n    const timeFraction = Math.min(elapsedTime / duration, 1);\n    // Apply the easing function to get the current progress\n    const easing = easeOutQuad(timeFraction);\n    // Calculate the current thermometer level using easing\n    const currentValue = startValue + changeInValue * easing;\n\n    // Update the thermometer level by calling setTemp\n    configuration.setTemp(currentValue);\n\n    // If the animation is not yet complete, request the next animation frame\n    if (timeFraction < 1) {\n      requestAnimationFrame(animate);\n    }\n  }\n\n  requestAnimationFrame(animate);\n  configuration.current = targetTemp;\n",
```